drivers/hmis/app/models/hmis/hud/custom_assessment.rb

```diff
@@ -155,6 +165,33 @@ def self.new_with_defaults(enrollment:, user:, form_definition:, assessment_date
     new_assessment
   end
 
+  def self.group_household_assessments(household_enrollments:, assessment_role:, threshold:, assessment_id: nil)
+    source_assessment = Hmis::Hud::CustomAssessment.find(assessment_id) if assessment_id.present?
+    # FIXME wont work for wip.
+    # raise HmisErrors::ApiError, 'Assessment not in household' if source_assessment.present? && !enrollments.pluck(:enrollment_id).include(source_assessment.enrollment_id)
+
+    household_assessments = Hmis::Hud::CustomAssessment.with_role(assessment_role.to_sym).for_enrollments(household_enrollments)
+
+    case assessment_role.to_sym
+    when :INTAKE, :EXIT
+      # Ensure we only return 1 assessment per person
+      household_assessments.group_by(&:personal_id).map { |_, a| a.last }
```

Comment:
any reason not to just:
```suggestion
      household_assessments.index_by(&:personal_id).values
```

Reply:
nope, thats better, thanks!